src/core/shader.js

```diff
@@ -60,9 +60,7 @@ Shader.prototype = {
     var self = this;
     var variables = {};
     var schema = this.schema;
-    var squemaKeys = Object.keys(schema);
-    squemaKeys.forEach(processSquema);
-    function processSquema (key) {
```

Comment:
I like having independent functions defined at the bottom. When you have multiple instructions makes the code more legible rather that having a foreign function embedded in the main code flow.

````javascript
var squemaKeys = Object.keys(schema);
squemaKeys.forEach(processSquema);
doThis();
doThat();
doMoreStuff();
function processSquema() { ... }
````